src/main/java/org/commcare/xml/StackFrameStepParser.java

```diff
@@ -22,17 +23,41 @@
     @Override
     public StackFrameStep parse() throws InvalidStructureException, IOException, XmlPullParserException {
         String operation = parser.getName();
+        String value = parser.getAttributeValue(null, "value");
 
         switch (operation) {
             case "datum":
-                String id = parser.getAttributeValue(null, "id");
-                return parseValue(SessionFrame.STATE_UNKNOWN, id);
+                String datumId = parser.getAttributeValue(null, "id");
+                return parseValue(SessionFrame.STATE_UNKNOWN, datumId);
             case "rewind":
                 return parseValue(SessionFrame.STATE_REWIND, null);
             case "mark":
                 return parseValue(SessionFrame.STATE_MARK, null);
             case "command":
                 return parseValue(SessionFrame.STATE_COMMAND_ID, null);
+            case "query":
+                String queryId = parser.getAttributeValue(null, "id");
+                String url = parser.getAttributeValue(null, "value");
```

Comment:
we should probably do some validation like [here](https://github.com/dimagi/commcare-core/blob/master/src/main/java/org/commcare/xml/SessionDatumParser.java#L98) on url

Reply:
Simon addressed in https://github.com/dimagi/commcare-core/pull/1048/commits/d1ea42af5e893cb33a097583fd40c21b89027ade